signer/src/transaction_coordinator.rs

```diff
@@ -1597,50 +1833,48 @@ where
         // Get the current sBTC limits (caps).
         let sbtc_limits = self.context.state().get_current_limits();
 
-        let context_window = self.context_window;
-        let accept_threshold = self.threshold;
-
-        // Fetch eligible deposit requests.
-        let pending_deposit_requests = storage
-            .get_pending_accepted_deposit_requests(
-                bitcoin_chain_tip.as_ref(),
-                context_window,
-                accept_threshold,
-            )
-            .await?;
-
-        // Fetch eligible withdrawal requests.
-        let pending_withdraw_requests = storage
-            .get_pending_accepted_withdrawal_requests(
-                bitcoin_chain_tip.as_ref(),
+        // Fetch pending deposit and withdrawal requests from storage.
+        let (deposits, withdrawals) = {
+            // Setup the parameters for fetching pending requests.
+            let params = GetPendingRequestsParams {
+                bitcoin_chain_tip,
                 stacks_chain_tip,
-                context_window,
-                accept_threshold,
+                aggregate_key,
+                signer_public_keys,
+                signature_threshold: self.threshold,
+                sbtc_limits: &sbtc_limits,
+            };
+
+            // Fetch eligible deposit requests.
+            let deposits =
+                Self::get_eligible_pending_deposit_requests(&storage, self.context_window, &params)
+                    .await?;
+
+            // Calculate the withdrawal expiry window, which is effectively how
+            // many canonical bitcoin blocks back in time we will look for
+            // eligible requests. We subtract the withdrawal expiry buffer from
+            // the withdrawal expiry period to ensure that we don't include
+            // requests that are about to expire. See the constant descriptions
+            // for more details.
+            let withdrawal_expiry_window = WITHDRAWAL_BLOCKS_EXPIRY
+                .saturating_sub(WITHDRAWAL_EXPIRY_BUFFER)
+                .try_into()
+                .map_err(|_| Error::TypeConversion)?;
```

Comment:
We can make the difference here it's own constant.

Reply:
Yeah, that works. Perfect.

Reply:
Added the const in [4d8eb68](https://github.com/stacks-network/sbtc/pull/1399/commits/4d8eb687e5c6e8b474f9c537c3b5991e1a247027) but it ended up being used in only one place 🙈

Reply:
That const is now removed as it was only used in one test